### ISSUE TITLE
Configurable proxy*interval

### ISSUE THREAD

**Issue by LordFPL** [sorintlab/stolon#314]
Hello,

Can it be possible to add a way to configure : 
```
DefaultProxyCheckInterval   = 5 * time.Second
DefaultProxyTimeoutInterval = 15 * time.Second
```
Or there is a reason to keep theses values ?

**Comment by sgotti** [comment]
@LordFPL Now they are hardcoded.

May I ask you the reason why you want to change them? (timeouts?).

They could probably be made configurable since #310 has been merged but we should add some checks (timeout > checkinterval) and put an upper bound since increasing them to too big values could increase the failover time.

**Comment by LordFPL** [comment]
Only because during my tests of failover, i modify "sleepInterval" to 1s... and was surprised to see proxy chek always on 5s because documentation say : 
```
sleepInterval | interval to wait before next check (for every component: keeper, sentinel, proxy). | no | string (duration) | 5s
```
I'm trying to see how i can reduce failover time.
Not a very big problem, 5s is already nice :)

**Comment by sgotti** [comment]
@LordFPL yeah, the doc is currently not right (we changed the proxy logic without fixing the doc).


> I'm trying to see how i can reduce failover time.

As a warning, now we are not checking all the values and their possible interaction (for example setting `failInterval` < `sleepInterval` will probably cause the keepers to be marked as failed if they're missing only a single check).

Also lowering the various intervals to too low ones can cause some unwanted failovers (for example if the machine is under load).

**Comment by trmfreitas** [comment]
I am doing this change but I am not validating values interaction.

**Comment by sgotti** [comment]
Implemented in #756.
